src/v/rpc/transport.cc

```diff
@@ -123,7 +123,14 @@ base_transport::connect(clock_type::time_point connection_timeout) {
 }
 ss::future<> base_transport::stop() {
     fail_outstanding_futures();
-    return _dispatch_gate.close();
+
+    // We must call stop() on our output stream, because seastar::output_stream
+    // may not be safely destroyed without a call to close(), and this class
+    // may be destroyed after stop() is called.
+    return _out.stop().then([this]() {
+        _out = {};
```

Comment:
I guess a comment here would be helpful since without going through all the PR is not obvious that default initialized batched output stream is used here as a safety check